normandy/recipes/filters.py

```diff
@@ -194,6 +194,76 @@ def capabilities(self):
         return set()
 
 
+class PrefFilter(BaseFilter):
+    """Match a user based on a preference having some sort of equality with a value.
+        Also, this filter can match based on if a pref is user set.
+
+    .. attribute:: type
+
+        ``pref``
+
+    .. attribute:: value
+
+        Can be a string representation of a number, characters, or boolean value.
+
+        :example: ``true`` or ``10`` or ``default``
```

Comment:
I'm not quite sure how to do this, but I think we should use the JSON type system instead of making everything a string. So instead of `{"value": "10"}`, `{"value": "true"}` and `{"value": "'default'"}`, we would have `{"value":10}`, `{"value": true}`, and `{"value": "default"}`. This makes it clearer what type is intended, and I think it will also avoid errors where we forget the quotes in the case the value is a string.

One important reason for this is that sometimes we can't tell if something should be a string or a number easily by looking at it. In [this rollout](https://experimenter.services.mozilla.com/experiments/doh-us-staged-rollout-to-all-us-desktop-users/) we needed to set a preference value to the *string* `"1896163212345"`. This is because that number is too large to fit in a preference integer without overflowing.

Again, I'm not sure how we can get the serializer to accept this, but I'm sure there is a way.